plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/BaseTrinoCatalogTest.java

```diff
@@ -48,40 +58,88 @@
 
     protected abstract TrinoCatalog createTrinoCatalog(boolean useUniqueTableLocations);
 
+    protected Map<String, Object> defaultNamespaceProperties(String newNamespaceName)
+    {
+        return ImmutableMap.of();
+    }
+
     @Test
     public void testCreateNamespaceWithLocation()
     {
         TrinoCatalog catalog = createTrinoCatalog(false);
-
         String namespace = "test_create_namespace_with_location_" + randomTableSuffix();
-        catalog.createNamespace(SESSION, namespace, ImmutableMap.of(LOCATION_PROPERTY, "/a/path/"), new TrinoPrincipal(PrincipalType.USER, SESSION.getUser()));
+        Map<String, Object> namespaceProperties = new HashMap<>(defaultNamespaceProperties(namespace));
+        String namespaceLocation = (String) namespaceProperties.computeIfAbsent(LOCATION_PROPERTY, ignored -> "/a/path/");
+        namespaceProperties = ImmutableMap.copyOf(namespaceProperties);
+        catalog.createNamespace(SESSION, namespace, namespaceProperties, new TrinoPrincipal(PrincipalType.USER, SESSION.getUser()));
         assertThat(catalog.listNamespaces(SESSION)).contains(namespace);
-        assertEquals(catalog.loadNamespaceMetadata(SESSION, namespace), ImmutableMap.of(LOCATION_PROPERTY, "/a/path/"));
-        assertEquals(catalog.defaultTableLocation(SESSION, new SchemaTableName(namespace, "table")), "/a/path/table");
+        assertEquals(catalog.loadNamespaceMetadata(SESSION, namespace), namespaceProperties);
+        assertEquals(catalog.defaultTableLocation(SESSION, new SchemaTableName(namespace, "table")), namespaceLocation.replaceAll("/$", "") + "/table");
         catalog.dropNamespace(SESSION, namespace);
         assertThat(catalog.listNamespaces(SESSION)).doesNotContain(namespace);
     }
 
     @Test
-    public void testCreateTable()
-            throws IOException
+    public void testNonLowercaseNamespace()
     {
         TrinoCatalog catalog = createTrinoCatalog(false);
-        Path tmpDirectory = Files.createTempDirectory("iceberg_catalog_test_create_table_");
-        tmpDirectory.toFile().deleteOnExit();
 
+        String namespace = "testNonLowercaseNamespace" + randomTableSuffix();
+        // Trino schema names are always lowercase (until https://github.com/trinodb/trino/issues/17)
+        String schema = namespace.toLowerCase(ENGLISH);
+
+        // Currently this is actually stored in lowercase by all Catalogs
```

Comment:
Is that true if you create the namespace via Spark?

Reply:
that's a property of the Glue/HMS, not the engine.
Note that TrinoCatalog implementations don't lowercase themselves